### ISSUE TITLE
Issue building TKW-X jar

### ISSUE THREAD

**Issue by j4jig** [nhsdigitalmait/TKW-x#7]
Hi Team,

We are trying to run the NEMS tests for our application to complete the accreditation process.
We downloaded TKW from https://digital.nhs.uk/services/interoperability-toolkit/developer-resources/itk-test-centre/itk-testbench but there are no domains folders for Event subscription API and NEMS message receiver tests.

We then find one git repository https://github.com/nhsdigitalmait . This repository looks like it has the configuration related to simulation for NEMS subscription and message validation. 

We have tried building TKW-x jar which in turn have multiple dependent projects such as Spine Tools ,JWT tools,common utils & DistributionEnvelope tools. 
Although we were able to get through the build of few of the dependencies. We are facing failures of many test cases in projects such as Spine Tools and hence we are unable to get the complete successful TKW-X.jar, which as per our understanding shall be used to run the simulation as a host for testing and creating validation reports for our system.

Could you  please provide the tkw-x.jar  or help us in understanding how we can resolve those dependencies while building the tkw-x git project so that we can proceed using TKW for testing our system.

For your reference, attaching some of the failures of Junit tests while building tkw-x project on our end:-
<img width="1651" alt="Screenshot 2020-12-13 at 2 43 15 AM" src="https://user-images.githubusercontent.com/31380857/101995706-30002480-3cf2-11eb-9c1b-ae358629fea0.png">
<img width="1671" alt="Screenshot 2020-12-13 at 2 43 24 AM" src="https://user-images.githubusercontent.com/31380857/101995711-35f60580-3cf2-11eb-8276-554d943599f6.png">
<img width="1664" alt="Screenshot 2020-12-13 at 12 52 08 AM" src="https://user-images.githubusercontent.com/31380857/101995712-37bfc900-3cf2-11eb-9c35-7ab0572ed2b9.png">
<img width="1675" alt="Screenshot 2020-12-13 at 12 52 29 AM" src="https://user-images.githubusercontent.com/31380857/101995713-38585f80-3cf2-11eb-85c4-0156f2a12873.png">

**Comment by rprobinson** [comment]
Hi,
Without the builds of these libraries (which are present in the nhsdigitalmait repo) you wont be able to build TKW-x. The repos here are snapshots of our development and not the best way to generate a working version of TKW + configurations for a specific programme. There is however a dockerhub repository at https://hub.docker.com/repository/docker/nhsdigitalmait/nems_subscriber_config. This will avoid all your setup issues and is the same code as is running on Opentest. There are full instructions in the comments section on that page.

**Comment by j4jig** [comment]
Hi Richard,

Thanks for your earlier response.

Based on your inputs, we were able to pull and run the docker image you pointed , in our system. We have been successful in running tests against our code for subscription API calls for NEMS events.

But we are now facing issues while running test for **verifyNHSNumber** call to SMSP PDS via TKW, which is a mandatory step to test prior calling the subscription Api. Here is the brief summary of the issues that we are facing with attached evidences for your reference :-

1. When calling **verifyNHSNumber** api at the port 8080 for TKW, we are getting the "Socket hang up" as shown below:
![postmanErrMsg_verifyNhsNumber](https://user-images.githubusercontent.com/31380857/103021828-d1983900-4570-11eb-8b82-bfdb37144e46.PNG)

This is the request that we are sending:
[Tkw-x_verifyNHSNumber_body.txt](https://github.com/nhsdigitalmait/TKW-x/files/5736350/Tkw-x_verifyNHSNumber_body.txt)

2. We have also tried placing explicitly the message to be validated in "messages_for_validation" folder and started the docker container to check if the message gets validated , but no logs and reports are generated. 

FYI, 
- We were able to validate the same message from the version of TKW downloaded from [here](https://digital.nhs.uk/services/interoperability-toolkit/developer-resources/itk-test-centre/itk-testbench).

- The message we place in message_for_validation is same message that we use in api call of verify NHS number in open test env which gives us the correct response.

Just wanted to clarify, what would be the right way to test the  verifyNHSNumber? Is there any other separate docker image that we can pull or we have to just validate the request message via the TKW downloaded from  [here](https://digital.nhs.uk/services/interoperability-toolkit/developer-resources/itk-test-centre/itk-testbench) ?
